src/types/app-factory.ts

```diff
@@ -458,6 +460,16 @@ export class AppFactory {
       this._clearSourceMap = result.compiledClear.sourceMap
     }
 
+    // Add source map persistence logic
+    if (Config.debug && Config.projectRoot) {
```

Comment:
Same comment about not needing `Config.projectRoot`

Reply:
Actually this function appears to call `AppClient.compile`. If i'm tracing things through correctly won't this even be emitted from there?